OPHD/UI/Core/RadioButtonGroup.h

```diff
@@ -0,0 +1,64 @@
+#pragma once
+
+#include "TextControl.h"
+#include "Label.h"
+#include "../../Cache.h"
+#include "../../Constants.h"
+
+#include <NAS2D/Signal/Signal.h>
+#include <NAS2D/Signal/Delegate.h>
+#include <NAS2D/EventHandler.h>
+#include <NAS2D/Resource/Image.h>
+#include <NAS2D/Utility.h>
+#include <NAS2D/Renderer/Renderer.h>
+#include <NAS2D/MathUtils.h>
+
+#include <algorithm>
+#include <string>
+
+class RadioButtonGroup : public Control
+{
+public:
+	struct ButtonInfo
+	{
+		std::string name;
+		NAS2D::DelegateX<void> delegate;
+	};
+
+	RadioButtonGroup() = default;
+	RadioButtonGroup(std::vector<ButtonInfo> buttonInfos);
+	void setSelected(std::size_t index);
+	void onMove(NAS2D::Vector<int> displacement) override;
+	void update() override;
+	void clearSelection();
+
+private:
+    class RadioButton : public TextControl
+    {
+    public:
+        RadioButton(RadioButtonGroup* parentContainer, std::string newText, NAS2D::DelegateX<void> delegate);
+        ~RadioButton() override;
+        void checked(bool toggle);
+        bool checked() const;
+        void text(const std::string& text);
+        const std::string& text() const;
+        void update() override;
+
+    protected:
+        void onResize() override;
+        void onTextChange() override;
+        void parentContainer(RadioButtonGroup* parent);
+        void onMouseDown(NAS2D::EventHandler::MouseButton button, int x, int y);
```

Comment:
We should probably move `parentContainer` out from the middle of the event handler methods. Actually, do we even need a `parentContainer` setter method anymore? Seems like that could/should be set once in the constructor and then never changed.

Reply:
Agreed, it's cleaner if it's only set through the constructor. The setter method has been removed.